### PR TITLE
fix(ens card): look up fallback && ui

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-2029
https://mask.atlassian.net/browse/MF-2099
https://mask.atlassian.net/browse/MF-2064
https://mask.atlassian.net/browse/MF-2103
https://mask.atlassian.net/browse/MF-2121

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/MaskBox/hooks/useContext.ts [hunk 1]
```diff
@@ -141,10 +144,13 @@ function useContext(initialState?: { boxId: string; hashRoot: string }) {
     const notInWhiteList = value?.message === 'leaf not found'
 
     // at least hold token amount
-    const { value: holderToken } = useFungibleToken(NetworkPluginID.PLUGIN_EVM, boxInfo?.holderTokenAddress)
+    const { value: holderToken } = useFungibleToken(NetworkPluginID.PLUGIN_EVM, boxInfo?.holderTokenAddress, {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/MaskBox/hooks/useContext.ts]: Use `<PluginIDContextProvider  />` and `<PluginWeb3ContextProvider />` stead.

Learn more: https://docs.mask.io/docs/web3/hooks

File: packages/plugin-infra/src/web3/useLookupAddress.ts [hunk 2]
```diff
@@ -11,9 +12,18 @@ export function useLookupAddress<T extends NetworkPluginID>(
...
+    const web3 = useWeb3(NetworkPluginID.PLUGIN_EVM)
 
     return useAsyncRetry(async () => {
-        if (!chainId || !domain || !Others?.isValidDomain?.(domain) || !NameService) return
-        return NameService.lookup?.(chainId, domain)
-    }, [chainId, domain, NameService, Others])
+        if (!domain || !Others?.isValidDomain?.(domain) || !web3) return
+        try {
+            if (!chainId || !NameService?.lookup) {
+                return web3.eth.ens.registry.getOwner(domain)
```
Comment (guanbinrui) [inline packages/plugin-infra/src/web3/useLookupAddress.ts]: The EVM chain-specific implementation should go here: `packages/plugins/EVM/src/state/NameService/ENS.ts` All Web3 hooks are shared across all networks.

File: packages/mask/src/plugins/MaskBox/contexts/RootContext.tsx [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
-import { TargetChainIdContext } from '@masknet/plugin-infra/web3-evm'
 
 interface Props {
     chainId: ChainId
 }
 
 export const RootContext: FC<PropsWithChildren<Props>> = ({ children, chainId }) => {
-    return <TargetChainIdContext.Provider initialState={chainId}>{children}</TargetChainIdContext.Provider>
+    return (
+        <PluginIDContextProvider value={NetworkPluginID.PLUGIN_EVM}>
```
Comment (guanbinrui) [inline packages/mask/src/plugins/MaskBox/contexts/RootContext.tsx:12]: So you don't need it in the ENS plugin anymore?

Comment (hanszhoou) [reply packages/mask/src/plugins/MaskBox/contexts/RootContext.tsx:12]: Added

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**